File: Lab/C++/DFS.cpp
#include <iostream>
#include <vector>

using namespace std;
int nrNoduri, nrMuchii, nodStart;
vector<int> vizitat;
vector<int> tata;
vector<int> muchii[1001];
void citire(){
    cin >> nrNoduri >> nrMuchii;
    for(int i = 0; i < nrNoduri; ++i){
        vizitat.push_back(0);
        tata.push_back(-1);
    }
    for(int i = 0; i < nrMuchii; ++i){
        int a,b;
        cin >> a >> b;
        muchii[a].push_back(b);
        muchii[b].push_back(a);
    }
}
void DFS(int x){
    vizitat[x] = 1;
    cout << x << " ";
    for(int i = 0;i < muchii[x].size(); ++i)
        if( vizitat[muchii[x][i]] == 0)
            DFS(muchii[x][i]);
}
int main(){
    citire();
    for(int i = 1;i <= nrNoduri; ++i)
        if( vizitat[i] == 0)
            DFS(i);
}

File: Lab/Python/DFS.py

f = open("graf.in")
line = f.readline()
v = line.split()
n = int(v[0])
m = int(v[1])

def citire_matrice_liste(noduri,muchii,orientat):
    l = {}
    for i in range(1, noduri + 1):
        l[i] = []

    for i in range(muchii):
        line = f.readline()
        v = line.split()
        x, y = v
        x = int(x)
        y = int(y)
        if orientat:
            l[x].append(y)
        else:
            l[x].append(y)
            l[y].append(x)
    return l

lista = citire_matrice_liste(n,m,0)
vizitat = [0] * (n + 1)
tata = [0] * (n + 1)

def DFS(nod):
    global vizitat, tata, lista
    vizitat[nod] = 1
    for y in lista[nod]:
        if vizitat[y] == 0:
            print(y, end = " ")
            tata[y] = nod
            DFS(y)

start = 1
DFS(start)

def listant(start):
    while start != 0:
        print(start, end = " ")
        start = tata[start]

listant(start)